samples/extauthz/cmd/extauthz/main.go

```diff
@@ -73,44 +73,50 @@ type ExtAuthzServer struct {
 	grpcPort chan int
 }
 
-// Check implements gRPC v2 check request.
-func (s *extAuthzServerV2) Check(ctx context.Context, request *authv2.CheckRequest) (*authv2.CheckResponse, error) {
-	l := fmt.Sprintf("%s%s, attributes: %v\n",
-		request.GetAttributes().GetRequest().GetHttp().GetHost(),
-		request.GetAttributes().GetRequest().GetHttp().GetPath(),
+func (s *extAuthzServerV2) logRequest(allow bool, request *authv2.CheckRequest) {
+	httpAttrs := request.GetAttributes().GetRequest().GetHttp()
+	allowStr := "allowed"
+	if !allow {
+		allowStr = "denied"
+	}
+	log.Printf("[gRPCv2][%s]: %s%s, attributes: %v\n", allowStr, httpAttrs.GetHost(),
+		httpAttrs.GetPath(),
 		request.GetAttributes())
-	if allowedValue == request.GetAttributes().GetRequest().GetHttp().GetHeaders()[checkHeader] || (request.GetAttributes().Source != nil && strings.HasSuffix(request.GetAttributes().Source.Principal, "/sa/"+*serviceAccount)) {
-		log.Printf("[gRPCv2][allowed]: %s", l)
-		return &authv2.CheckResponse{
-			HttpResponse: &authv2.CheckResponse_OkResponse{
-				OkResponse: &authv2.OkHttpResponse{
-					Headers: []*corev2.HeaderValueOption{
-						{
-							Header: &corev2.HeaderValue{
-								Key:   resultHeader,
-								Value: resultAllowed,
-							},
+}
+
+func (s *extAuthzServerV2) allow(request *authv2.CheckRequest) *authv2.CheckResponse {
+	s.logRequest(true, request)
```

Comment:
I think you can just pass the `allowStr string` directly, and `resultAllowed`/`resultDenied` is also defined so can be used directly.